src/DynamicDataTable.jsx

```diff
@@ -524,7 +524,11 @@ class DynamicDataTable extends Component {
             </div>
         );
 
-        if (value === 'all' && this.props.renderMasterCheckbox) {
+        if (value === 'all') {
+            if (this.props.renderMasterCheckbox) {
```

Comment:
I think the if-statement should be inverted.

```suggestion
            if (!this.props.renderMasterCheckbox) {
```